stable/database/templates/_helpers.tpl

```diff
@@ -172,15 +172,35 @@ Get security context runAsUser and runAsGroup
 {{- if eq (include "defaultfalse" .Values.database.securityContext.enabled) "true" }}
 runAsUser: {{ default 1000 .Values.database.securityContext.runAsUser }}
 runAsGroup: 0
-  {{- if and (or (eq (include "defaulttrue" .Values.database.initContainers.runInitDisk) "false") (eq (include "defaulttrue" .Values.database.initContainers.runInitDiskAsRoot) "false")) (ne (toString (default 1000 .Values.database.securityContext.runAsUser)) "0") }}
-runAsNonRoot: true
-  {{- end }}
+{{- if ne (toString (default 1000 .Values.database.securityContext.runAsUser)) "0" }}
+{{- include "sc.runAsNonRoot" . }}
+{{- end }}
 {{- else if eq (include "defaultfalse" .Values.database.securityContext.runAsNonRootGroup) "true" }}
 runAsUser: 1000
 runAsGroup: 1000
-  {{- if or (eq (include "defaulttrue" .Values.database.initContainers.runInitDisk) "false") (eq (include "defaulttrue" .Values.database.initContainers.runInitDiskAsRoot) "false") }}
+{{- include "sc.runAsNonRoot" . }}
+{{- end }}
+{{- end -}}
+
+{{/*
+Get security context runAsNonRoot
+*/}}
+{{- define "sc.runAsNonRoot" -}}
+{{- $runAsNonRoot := true -}}
+{{- if and
+       (eq (include "defaulttrue" .Values.database.initContainers.runInitDisk) "true")
+       (eq (include "defaulttrue" .Values.database.initContainers.runInitDiskAsRoot) "true") }}
+  {{- $runAsNonRoot = false -}}
+{{- else if .Values.database.backupHooks -}}
+  {{- if and
+         (eq (include "defaultfalse" .Values.database.backupHooks.enabled) "true")
+         (eq (include "defaultfalse" .Values.database.backupHooks.useSuspend) "false")
+         (eq (include "defaultfalse" .Values.database.sm.noHotCopy.journalPath.enabled) "true") -}}
```

Comment:
Technically this is not needed for the TE deployments as they don't have backup hooks container but I guess it will be too much of a hassle to conditionalize the helper with _context_.

Reply:
True. I did not think the extra complexity was warranted just to decouple the security contexts from each other. If there's a policy that requires `runAsNonRoot=true`, we cannot be in compliance with it anyway if fsfreeze is required, so I'm not sure what the value is making the TE Deployment compliant by itself.